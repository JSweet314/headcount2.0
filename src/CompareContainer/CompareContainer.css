.Compared-container {
  display: flex;
  width: 100%;
}

#left {
  order: 0;
}

#right {
  order: 2
}




