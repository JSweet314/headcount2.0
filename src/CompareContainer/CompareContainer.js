import React from 'react';
import Card from '../Card/Card.js';
import PropTypes from 'prop-types';

const CompareContainer = ({searchValue, selectLocation, selectedLocations}) => {
	const comparedDistricts = selectedLocations.map(
		(district, index) => { 
		return <Card {...district} 
			key={index}
			className={index < 1 ? 'left' : 'right'}
			searchValue={searchValue} 
      selectLocation={selectLocation}
			selectedLocations={selectedLocations} />
	});

	return (
		<div className="Compared-container">
			{comparedDistricts}
		</div>
	);
}

CompareContainer.propTypes = {
	searchValue: PropTypes.string.isRequired,
	selectLocation: PropTypes.func.isRequired,
	selectedLocations: PropTypes.array.isRequired
};

export default CompareContainer;