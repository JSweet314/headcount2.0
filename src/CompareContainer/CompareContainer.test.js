import React from 'react';
import {shallow} from 'enzyme';
import CompareContainer from './CompareContainer.js';

describe( 'CompareContainer', () => {
  const selectLocation = jest.fn();
  const comparedAverage = {
    COLORADO: 0.32,
    'COLORADO SPRINGS': 0.76,
    compared: 0.5 
  };

  let wrapper, selectedLocations;

  beforeEach(()=> {
    selectedLocations = [];
    wrapper = shallow(
      <CompareContainer
        searchValue={''}
        selectLocation={selectLocation} 
        selectedLocations={selectedLocations}
        comparedAverage={comparedAverage} 
      />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render the Card when there are 2 selected locations', () => {
    selectedLocations = [{ 
      location: 'COLORADO', 
      stats: {
        '2017': 0.34,
        '2016': 0.45
      } 
    }, { 
      location: 'COLORADO SPRINGS', 
      stats: {
        '2017': 0.34,
        '2016': 0.45
      } 
    }];
    wrapper = shallow(
      <CompareContainer
        searchValue={''}
        selectLocation={selectLocation} 
        selectedLocations={selectedLocations}
        comparedAverage={comparedAverage} 
      />);
    expect(wrapper).toMatchSnapshot();

  });
});

