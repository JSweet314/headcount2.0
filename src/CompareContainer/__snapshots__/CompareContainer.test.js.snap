// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareContainer Should render the Card when there are 2 selected locations 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareContainer
    comparedAverage={
        Object {
            "COLORADO": 0.32,
            "COLORADO SPRINGS": 0.76,
            "compared": 0.5,
          }
    }
    searchValue=""
    selectLocation={[Function]}
    selectedLocations={
        Array [
            Object {
              "location": "COLORADO",
              "stats": Object {
                "2016": 0.45,
                "2017": 0.34,
              },
            },
            Object {
              "location": "COLORADO SPRINGS",
              "stats": Object {
                "2016": 0.45,
                "2017": 0.34,
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <Card
            className="left"
            location="COLORADO"
            searchValue=""
            selectLocation={[Function]}
            selectedLocations={
                        Array [
                                    Object {
                                      "location": "COLORADO",
                                      "stats": Object {
                                        "2016": 0.45,
                                        "2017": 0.34,
                                      },
                                    },
                                    Object {
                                      "location": "COLORADO SPRINGS",
                                      "stats": Object {
                                        "2016": 0.45,
                                        "2017": 0.34,
                                      },
                                    },
                                  ]
            }
            stats={
                        Object {
                                    "2016": 0.45,
                                    "2017": 0.34,
                                  }
            }
/>,
          <Card
            className="right"
            location="COLORADO SPRINGS"
            searchValue=""
            selectLocation={[Function]}
            selectedLocations={
                        Array [
                                    Object {
                                      "location": "COLORADO",
                                      "stats": Object {
                                        "2016": 0.45,
                                        "2017": 0.34,
                                      },
                                    },
                                    Object {
                                      "location": "COLORADO SPRINGS",
                                      "stats": Object {
                                        "2016": 0.45,
                                        "2017": 0.34,
                                      },
                                    },
                                  ]
            }
            stats={
                        Object {
                                    "2016": 0.45,
                                    "2017": 0.34,
                                  }
            }
/>,
        ],
        <ComparisonCard
          comparedAverage={
                    Object {
                              "COLORADO": 0.32,
                              "COLORADO SPRINGS": 0.76,
                              "compared": 0.5,
                            }
          }
/>,
      ],
      "className": "Compared-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "className": "left",
          "location": "COLORADO",
          "searchValue": "",
          "selectLocation": [Function],
          "selectedLocations": Array [
            Object {
              "location": "COLORADO",
              "stats": Object {
                "2016": 0.45,
                "2017": 0.34,
              },
            },
            Object {
              "location": "COLORADO SPRINGS",
              "stats": Object {
                "2016": 0.45,
                "2017": 0.34,
              },
            },
          ],
          "stats": Object {
            "2016": 0.45,
            "2017": 0.34,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "className": "right",
          "location": "COLORADO SPRINGS",
          "searchValue": "",
          "selectLocation": [Function],
          "selectedLocations": Array [
            Object {
              "location": "COLORADO",
              "stats": Object {
                "2016": 0.45,
                "2017": 0.34,
              },
            },
            Object {
              "location": "COLORADO SPRINGS",
              "stats": Object {
                "2016": 0.45,
                "2017": 0.34,
              },
            },
          ],
          "stats": Object {
            "2016": 0.45,
            "2017": 0.34,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "comparedAverage": Object {
            "COLORADO": 0.32,
            "COLORADO SPRINGS": 0.76,
            "compared": 0.5,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <Card
              className="left"
              location="COLORADO"
              searchValue=""
              selectLocation={[Function]}
              selectedLocations={
                            Array [
                                          Object {
                                            "location": "COLORADO",
                                            "stats": Object {
                                              "2016": 0.45,
                                              "2017": 0.34,
                                            },
                                          },
                                          Object {
                                            "location": "COLORADO SPRINGS",
                                            "stats": Object {
                                              "2016": 0.45,
                                              "2017": 0.34,
                                            },
                                          },
                                        ]
              }
              stats={
                            Object {
                                          "2016": 0.45,
                                          "2017": 0.34,
                                        }
              }
/>,
            <Card
              className="right"
              location="COLORADO SPRINGS"
              searchValue=""
              selectLocation={[Function]}
              selectedLocations={
                            Array [
                                          Object {
                                            "location": "COLORADO",
                                            "stats": Object {
                                              "2016": 0.45,
                                              "2017": 0.34,
                                            },
                                          },
                                          Object {
                                            "location": "COLORADO SPRINGS",
                                            "stats": Object {
                                              "2016": 0.45,
                                              "2017": 0.34,
                                            },
                                          },
                                        ]
              }
              stats={
                            Object {
                                          "2016": 0.45,
                                          "2017": 0.34,
                                        }
              }
/>,
          ],
          <ComparisonCard
            comparedAverage={
                        Object {
                                    "COLORADO": 0.32,
                                    "COLORADO SPRINGS": 0.76,
                                    "compared": 0.5,
                                  }
            }
/>,
        ],
        "className": "Compared-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "className": "left",
            "location": "COLORADO",
            "searchValue": "",
            "selectLocation": [Function],
            "selectedLocations": Array [
              Object {
                "location": "COLORADO",
                "stats": Object {
                  "2016": 0.45,
                  "2017": 0.34,
                },
              },
              Object {
                "location": "COLORADO SPRINGS",
                "stats": Object {
                  "2016": 0.45,
                  "2017": 0.34,
                },
              },
            ],
            "stats": Object {
              "2016": 0.45,
              "2017": 0.34,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "className": "right",
            "location": "COLORADO SPRINGS",
            "searchValue": "",
            "selectLocation": [Function],
            "selectedLocations": Array [
              Object {
                "location": "COLORADO",
                "stats": Object {
                  "2016": 0.45,
                  "2017": 0.34,
                },
              },
              Object {
                "location": "COLORADO SPRINGS",
                "stats": Object {
                  "2016": 0.45,
                  "2017": 0.34,
                },
              },
            ],
            "stats": Object {
              "2016": 0.45,
              "2017": 0.34,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "comparedAverage": Object {
              "COLORADO": 0.32,
              "COLORADO SPRINGS": 0.76,
              "compared": 0.5,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CompareContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareContainer
    comparedAverage={
        Object {
            "COLORADO": 0.32,
            "COLORADO SPRINGS": 0.76,
            "compared": 0.5,
          }
    }
    searchValue=""
    selectLocation={[Function]}
    selectedLocations={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [],
        false,
      ],
      "className": "Compared-container",
    },
    "ref": null,
    "rendered": Array [
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [],
          false,
        ],
        "className": "Compared-container",
      },
      "ref": null,
      "rendered": Array [
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
